Feature/lp 956 update php sdk

alejouribesanchez opened the pull request:
- Added BankStatement process documents and GET endpoints
- Added AnyDocs process documents and GET endpoints
- Added W2 process documents and GET endpoints
- Added some unit tests
- Added missing PHP types for the functions
- Added support for PHP 8.3
- Update missing unit tests
- Update to version 1.0.4

alejouribesanchez committed:
Add support for new endpoints

Added BankStatement process documents and GET endpoints
Added AnyDocs process documents and GET endpoints
Added W2 process documents and GET endpoints
Added some unit tests
Added missing PHP types for the functions
Added support for PHP 8.3

alejouribesanchez committed:
Add missing unit tests

Update missing unit tests
Update to version 1.0.4

manycoding commented on src/veryfi/Client.php:
```diff
@@ -550,4 +551,297 @@ public function replace_tags(int $document_id,
+     * @return string An array of JSON with all W2 documents.
+     * @throws Exception when API version is not supported for W2 documents.
+     */
+    public function get_w2_documents(int $page = null): string
```
add additional parameters (kwargs in python), e.g. we have created_date__gte and etc., same for all other get_..s

alejouribesanchez replied:
added additional parameters as parameter to all GET endpoints

manycoding commented on src/veryfi/Client.php:
```diff
@@ -550,4 +551,297 @@ public function replace_tags(int $document_id,
+     * @param string $document_id The ID of the document you'd like to retrieve.
+     * @return string Data extracted from the document.
+     */
+    public function get_w2_document(string $document_id): string
```
is it string though?

alejouribesanchez replied:
Yes it is the expected output type

manycoding commented on src/veryfi/Client.php:
```diff
@@ -550,4 +551,297 @@ public function replace_tags(int $document_id,
+     * @return string Data extracted from the document.
+     */
+    public function process_w2_document(string $file_path,
+                                        bool $delete_after_processing = false,
```
let's use the same naming as in API - auto_delete

alejouribesanchez replied:
Done

manycoding commented on src/veryfi/Client.php:
```diff
@@ -550,4 +551,297 @@ public function replace_tags(int $document_id,
+        $endpoint_name = '/w2s/';
+        $file_name = pathinfo($file_path, PATHINFO_BASENAME);
+        $file = fopen($file_path, 'rb');
+        $file_data = base64_encode(fread($file, filesize($file_path)));
```
did you test it produces correct string with type hint `data:image/jpeg;....`?

alejouribesanchez replied:
Yes everything is tested against API and working well

manycoding commented on src/veryfi/Client.php:
```diff
@@ -550,4 +551,297 @@ public function replace_tags(int $document_id,
+     * @param array $additional_request_parameters Additional request parameters
+     * @return string Data extracted from the document
+     */
+    public function process_any_document_from_file(string $file_path, string $template_name, int $max_pages_to_process = 20, array $additional_request_parameters = []): string
```
how it's different from process_any_document?

alejouribesanchez replied:
1.	File Name Extraction:
	•	First Snippet: Uses basename($file_path) to extract the file name.
	•	Second Snippet: Uses pathinfo($file_path, PATHINFO_BASENAME) to extract the file name.
Effect: Both methods return the base name of the file, so functionally they are identical. However, pathinfo() is more versatile as it can extract other file path components (like file extension) if needed.
	2.	File Handling:
	•	First Snippet: Uses file_get_contents($file_path) to read the file in one go and encode it in base64.
	•	Second Snippet: Uses fopen() to open the file, then reads it using fread(), and encodes the content in base64.
Effect:
	•	file_get_contents() is simpler and reads the entire file at once. It’s easier to use, but can consume more memory for very large files since it loads everything into memory at once.
	•	fopen() with fread() is more granular and might give you more control over memory usage. However, for very large files, this approach can be more efficient as it allows for buffered reading (although here the entire file is still read into memory).

alejouribesanchez replied:
Done I removed the function who uses  file_get_contents() because it can consume more memory for very large files since it loads everything into memory at once

manycoding commented on src/veryfi/Client.php:
```diff
@@ -550,4 +551,297 @@ public function replace_tags(int $document_id,
+     * @param array $additional_request_parameters Additional request parameters
+     * @return string Data extracted from the document
+     */
+    public function process_bank_statement_from_file(string $file_path, bool $bounding_boxes = false, bool $confidence_details = false, array $additional_request_parameters = []): string
```
same as process_bank_statement?

alejouribesanchez replied:
Done I removed the function who uses file_get_contents() because it can consume more memory for very large files since it loads everything into memory at once

manycoding commented on src/veryfi/Client.php:
```diff
@@ -550,4 +551,297 @@ public function replace_tags(int $document_id,
+     * @param int|null $page The page number, response is capped to a maximum of 50 results per page.
+     * @return string An array of JSON with all W2 documents.
+     * @throws Exception when API version is not supported for W2 documents.
+     */
```
Let's add a link to this particular endpoint in docs here and for all others
https://docs.veryfi.com/api/w2s/get-w-2-s/

alejouribesanchez committed:
Add documentation

Add some links to the official documentation
Add missing additional arguments to the GET endpoints

alejouribesanchez committed:
Remove duplicate function

keep pathinfo() is more versatile as it can extract other file path components (like file extension) if needed. Remove  file_get_contents() is simpler and reads the entire file at once. It’s easier to use, but can consume more memory for very large files since it loads everything into memory at once

alejouribesanchez committed:
Adding hint to base64

alejouribesanchez committed:
Refactor SDK using Traits

alejouribesanchez committed:
Add missing requests

alejouribesanchez committed:
Add missing functions

alejouribesanchez committed:
Update unit tests

manycoding reviewed (changes requested):
api/v8/contracts

Kaevan89 commented:
our GET methods don't have body parameters, they are query parameters.

Kaevan89 reviewed (changes requested):
Fix GET methods with query parameters.

alejouribesanchez committed:
Add granular client tests and update signature logic

Split ClientTest.php into multiple focused test classes for documents, checks, bank statements, business cards, line items, tags, and general client functionality. Updated GenerateSignature trait to handle CURLFile and object payloads more robustly. Upgraded phpunit/phpunit to version 12.3.0 in composer.json.

alejouribesanchez committed:
Update CI workflows to use PHP 8.3

Replaces PHP versions 7.4, 8.0, 8.1, and 8.2 with 8.3 in main.yml and php.yml workflows. Also updates the code coverage commit condition to match PHP 8.3.

alejouribesanchez committed:
Update test imports to use correct Client class

Changed all test files to import 'veryfi\Client' instead of 'veryfi\client' to match the correct class name and adhere to PSR standards. This improves consistency and prevents potential autoloading issues.

alejouribesanchez committed:
Update phpunit.xml for PHPUnit 10 compatibility

Migrated phpunit.xml configuration to support PHPUnit 10.x, updating schema location, adding bootstrap and cacheDirectory, and restructuring coverage and source sections for improved test and coverage management.

alejouribesanchez committed:
Add ClientCoverageTest for line items and signature

Introduces ClientCoverageTest to verify initialization and validation of LineItem and LineItemUpdate classes, as well as signature generation in the Client class. Ensures proper exception handling and coverage for various payload types.

alejouribesanchez committed:
Add file-based tests for business cards, checks, and documents

Introduces new test methods for processing business cards, checks, and documents using file paths instead of base64. Existing base64 test methods are renamed for clarity. This improves test coverage and distinguishes between file and base64 input scenarios.
